tools/GenerateXdmf.py

```diff
@@ -46,49 +46,85 @@ def generate_xdmf(file_prefix, output_filename, start_time, stop_time, stride):
-                xdmf_output += data_item + element_path + \
-                    "/InertialCoordinates_x\n        </DataItem>\n"
-                xdmf_output += data_item + element_path + \
-                    "/InertialCoordinates_y\n        </DataItem>\n"
-                xdmf_output += data_item + element_path + \
-                    "/InertialCoordinates_z\n        </DataItem>\n"
-                xdmf_output += "      </Geometry>\n"
-                # Write tensor components as scalars
-                components = list(h5temporal.get(grid).keys())
-                components.remove('InertialCoordinates_x')
-                components.remove('InertialCoordinates_y')
-                components.remove('InertialCoordinates_z')
-                components.remove('connectivity')
-                for component in components:
+                    % (numpoints)
+            data_item_vec = "        <DataItem Dimensions=\" %d 3\" "\
+                    "ItemType = \"Function\" Function = \"JOIN($0,$1,$2)\">\n"\
+                            % (numpoints)
+            Grid_path = "          %s:/element_data.vol/%s" % (
+                h5file[1], id_and_value[0])
+            xdmf_output += \
+                           "    <Grid Name=\"%s\" GridType=\"Uniform\">\n" \
+                           % (h5file[1])
+            # Write topology information
+            xdmf_output += "      <Topology TopologyType=\"Hexahedron\" " \
+                           "NumberOfElements=\"%d\">\n" % (number_of_cells)
+            xdmf_output += "        <DataItem Dimensions=\"%d 8\" " \
+                           "NumberType=\"Int\" Format=\"HDF5\">\n" % (
+                               number_of_cells)
+            xdmf_output += Grid_path  + "/connectivity\n" \
+                           "        </DataItem>\n      </Topology>\n"
+            # Write geometry/coordinates
+            xdmf_output += "      <Geometry Type=\"X_Y_Z\">\n"
+            xdmf_output += data_item + Grid_path + \
+                            "/InertialCoordinates_x\n        </DataItem>\n"
+            xdmf_output += data_item + Grid_path + \
+                           "/InertialCoordinates_y\n        </DataItem>\n"
+            xdmf_output += data_item + Grid_path + \
+                           "/InertialCoordinates_z\n        </DataItem>\n"
+            xdmf_output += "      </Geometry>\n"
+            # Everything that isn't a coordinate is a "component"
+            components = list(h5temporal.keys())
+            components.remove('InertialCoordinates_x')
+            components.remove('InertialCoordinates_y')
+            components.remove('InertialCoordinates_z')
+            components.remove('connectivity')
+            components.remove('total_extents')
+            components.remove('grid_names')
+            for component in components:
+                # If the component is x component of a vector, write a
```

Comment:
We think it would be clearer to have:
```python
if component.endswith("_x"):
    # Write a vector using the three components that make up the vector (i.e. v_x, v_y, v_z)
```